File: snippets/node/incoming-payment/incoming-payment-get-public.ts
import dotenv from "dotenv";
import { join } from "path";
import { fileURLToPath } from "url";

dotenv.config({
    path: fileURLToPath(join(import.meta.url, "..", "..", ".env")),
});

const INCOMING_PAYMENT_URL = process.env.INCOMING_PAYMENT_URL;

//@! start chunk 1 | title=Import dependencies
import { createUnauthenticatedClient } from "@interledger/open-payments";
//@! end chunk 1

//@! start chunk 2 | title=Initialize Open Payments client
const client = await createUnauthenticatedClient({});
//@! end chunk 2

//@! start chunk 3 | title=Get incoming payment
const incomingPayment = await client.incomingPayment.get({
    url: INCOMING_PAYMENT_URL,
});
//@! end chunk 3

//@! start chunk 4 | title=Output
console.log("INCOMING PAYMENT:", incomingPayment);
//@! end chunk 4

File: snippets/rust/incoming-payment/incoming-payment-get-public.rs
//@! start chunk 1 | title=Import dependencies
use open_payments::client::api::UnauthenticatedResources;
use open_payments::snippets::utils::{create_unauthenticated_client, get_env_var, load_env};
//@! end chunk 1

#[tokio::main]
async fn main() -> open_payments::client::Result<()> {
    load_env()?;

    //@! start chunk 2 | title=Initialize Open Payments client
    let client = create_unauthenticated_client();
    //@! end chunk 2

    //@! start chunk 3 | title=Get without authentication
    let incoming_payment_url = get_env_var("INCOMING_PAYMENT_URL")?;

    let payment = client
        .public_incoming_payments()
        .get(&incoming_payment_url)
        .await?;
    //@! end chunk 3

    //@! start chunk 4 | title=Output
    println!("Public incoming payment: {payment:#?}");
    //@! end chunk 4
    Ok(())
}


